src/Ocelot/Configuration/DownstreamRouteExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Globalization;
+using System.Numerics;
+using System.Text.Json;
+
+namespace Ocelot.Configuration
+{
+    public static class DownstreamRouteExtensions
```

Comment:
I haven't checked it, but I think the logic could be simplified here, that would be an approach I would follow - but I haven't tested it yet -
```c#
public static class DownstreamRouteExtensions
{
    private static readonly HashSet<string> TruthyValues = new(StringComparer.OrdinalIgnoreCase) { "true", "yes", "on", "ok", "enable", "enabled", "1" };
    private static readonly string[] Separator = [","];

    public static T GetMetadata<T>(this DownstreamRoute downstreamRoute, string key, T defaultValue = default, CultureInfo cultureInfo = null)
    {
        var metadata = downstreamRoute?.Metadata;

        if (metadata == null || !metadata.TryGetValue(key, out var metadataValue))
        {
            return defaultValue;
        }

        return (T) ConvertTo(typeof(T), metadataValue, cultureInfo ?? CultureInfo.CurrentCulture);
    }

    private static object ConvertTo(Type targetType, string value, IFormatProvider cultureInfo)
    {
        if (targetType == typeof(string))
        {
            return value;
        }

        if (targetType == typeof(bool))
        {
            return TruthyValues.Contains(value.Trim());
        }

        if (targetType == typeof(string[]))
        {
            return value.Split(Separator, StringSplitOptions.RemoveEmptyEntries);
        }

        if (typeof(JsonSerializerContext).IsAssignableFrom(targetType))
        {
            return JsonSerializer.Deserialize(value, targetType);
        }

        try
        { 
            return Convert.ChangeType(value, targetType, cultureInfo);
        }
        catch (InvalidCastException)
        {
            return null;
        }
    }
}
```

Reply:
Thanks! It looks much simpler. I'll verify it soon.

Reply:
```C#
    private static readonly HashSet<string> TruthyValues = new(StringComparer.OrdinalIgnoreCase) { "true", "yes", "on", "ok", "enable", "enabled", "1" };
```
What about falsie values aka `false`?

Reply:
Do you have special conversion extension method for arrays?

Reply:
@vantm Where are you?